test/extended/poddisruptionbudget/poddisruptionbudget.go

```diff
@@ -0,0 +1,318 @@
+			}
+			pdb, err = waitForPDBStatus(oc.KubeClient().PolicyV1(), metav1.ObjectMeta{Name: ifHealthyBudgetPDBName, Namespace: oc.Namespace()}, e2e.PodStartTimeout, expectedStatus)
+			if err != nil {
+				g.Fail(fmt.Sprintf("error occurred while waiting for PDB status: %v\nexpected %#v, got %#v", err, expectedStatus, pdbStatus(pdb)))
+			}
+
+			g.By("eviction of the new running but not ready pod should succeed")
+			thirdNewToEvictPod := pods[0]
+			eviction = newV1Eviction(oc.Namespace(), thirdNewToEvictPod, metav1.DeleteOptions{})
+			err = oc.KubeClient().PolicyV1().Evictions(oc.Namespace()).Evict(context.Background(), eviction)
+			o.Expect(err).NotTo(o.HaveOccurred())
+			shouldTerminatePods.Insert(thirdNewToEvictPod)
+
+			g.By("wait until all evicted pods are either gone or terminating")
+			pods, err = exutil.WaitForPods(oc.KubeClient().CoreV1().Pods(oc.Namespace()), podsLabelSelector, func(pod v1.Pod) bool {
+				return shouldTerminatePods.Has(pod.Name) && pod.DeletionTimestamp == nil
+			}, 0, e2e.PodDeleteTimeout)
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			g.By("wait for PDB to become stable again")
+			expectedStatus = policyv1.PodDisruptionBudgetStatus{
+				DisruptionsAllowed: 1,
+				CurrentHealthy:     3,
+				DesiredHealthy:     2,
+				ExpectedPods:       3,
+			}
+			pdb, err = waitForPDBStatus(oc.KubeClient().PolicyV1(), metav1.ObjectMeta{Name: ifHealthyBudgetPDBName, Namespace: oc.Namespace()}, e2e.PodStartTimeout+podBecomesReadyTimeout, expectedStatus)
+			if err != nil {
+				g.Fail(fmt.Sprintf("error occurred while waiting for PDB status: %v\nexpected %#v, got %#v", err, expectedStatus, pdbStatus(pdb)))
+			}
+		})
+
+		g.It(fmt.Sprintf("should evict according to the AlwaysAllow policy"), func() {
+			g.By(fmt.Sprintf("calling oc create -f %q", alwaysAllowPolicyPDB))
+			err := oc.Run("create").Args("-f", alwaysAllowPolicyPDB).Execute()
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			g.By(fmt.Sprintf("calling oc create -f %q", nginxWithDelayedReadyDeployment))
+			err = oc.Run("create").Args("-f", nginxWithDelayedReadyDeployment).Execute()
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			g.By("wait for a PDB to initialize its status")
+			expectedStatus := policyv1.PodDisruptionBudgetStatus{
+				DisruptionsAllowed: 0,
+				CurrentHealthy:     0,
+				DesiredHealthy:     2,
+				ExpectedPods:       3,
+			}
+			pdb, err := waitForPDBStatus(oc.KubeClient().PolicyV1(), metav1.ObjectMeta{Name: alwaysAllowPolicyPDBName, Namespace: oc.Namespace()}, e2e.PodStartTimeout, expectedStatus)
+			if err != nil {
+				g.Fail(fmt.Sprintf("error occurred while waiting for PDB status: %v\nexpected %#v, got %#v", err, expectedStatus, pdbStatus(pdb)))
+			}
+
+			g.By("wait for pods to be running")
+			pods, err := exutil.WaitForPods(oc.KubeClient().CoreV1().Pods(oc.Namespace()), podsLabelSelector, exutil.CheckPodIsRunning, 3, e2e.PodStartTimeout)
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			g.By("evict unready pods guarded by a PDB")
```

Comment:
Is the diff b/w "ifHealthyBudgetPolicy" and "AlwaysAllwPolicy "is in the later one unready pods guarded by a PDB would get evicted ?

Reply:
yes, AlwaysAllow can evict all unready pods regardless of the PDB constraints (maxUnavailable, minAvailable) as compared to the IfHealthyBudget that follows these constraints. For more details see https://kubernetes.io/docs/tasks/run-application/configure-pdb/#unhealthy-pod-eviction-policy